ui/main/src/app/modules/usercard/usercard.component.ts

```diff
@@ -81,123 +109,222 @@ export class UserCardComponent implements OnDestroy {
+
+    }
+
+    reinsertScripts(): void {
+        const scripts = <HTMLScriptElement[]>this.element.nativeElement.getElementsByTagName('script');
+        const scriptsInitialLength = scripts.length;
+        for (let i = 0; i < scriptsInitialLength; i++) {
+            const script = scripts[i];
+            const scriptCopy = document.createElement('script');
+            scriptCopy.type = script.type ? script.type : 'text/javascript';
+            if (script.innerHTML) {
+                scriptCopy.innerHTML = script.innerHTML;
+            }
+            scriptCopy.async = false;
+            script.parentNode.replaceChild(scriptCopy, script);
         }
     }
 
+
+
     onSubmitForm(template: TemplateRef<any>) {
         const formValue = this.messageForm.value;
-
         const processFormVal = formValue['process'];
-        const selectedProcess = this.fetchedProcesses.find(process => {
+        const selectedProcess = this.processesDefinition.find(process => {
             return process.id === processFormVal;
         });
         const processVersion = selectedProcess.version;
-        const formValueElement = formValue['state'];
-        const selectedState = selectedProcess.states[formValueElement];
-        const titleKey = (new I18n((selectedState.name) ? selectedProcess.name : formValueElement));
-        const now = new Date().getTime();
+        const state = formValue['state'];
+
+        if (!templateGateway.getSpecificCardInformation) {
+            console.log('No getSpecificCardInformationMethod() in template can not send card');
+            this.errorMessage.display = true;
+            this.errorMessage.text = 'userCard.error.templateError';
+            return;
+        }
+
+        const specificInformation = templateGateway.getSpecificCardInformation();
+        if (!specificInformation) {
+            console.log('getSpecificCardInformationMethod() in template return no information');
+            this.errorMessage.display = true;
+            this.errorMessage.text = 'userCard.error.templateError';
+            return;
+        }
+
+        if (!specificInformation.valid) {
+            this.errorMessage.display = true;
+            this.errorMessage.text = specificInformation.errorMsg;
+            return;
+        }
+
+        const entities = new Array();
+        if (formValue['entities'].length < 1) {
+            this.errorMessage.display = true;
+            this.errorMessage.text = 'userCard.error.noRecipientSelected';
+            return;
+        } else {
+            formValue['entities'].forEach(entity => entities.push(entity.id));
+        }
+
+        let startDate = this.messageForm.get('startDate').value;
+        if ((!startDate) || startDate.value === '') startDate = this.defaultStartDate;
+
+        let endDate = this.messageForm.get('endDate').value;
+        if ((!endDate) || endDate.value === '') endDate = this.defaultEndDate;
```

Comment:
`endDate.value === ''`same as above, it could be `!endDate.value`

Reply:
done